### ISSUE TITLE
conv2d_extractor의 transpose에 관한 질문

### ISSUE THREAD

**Issue by Klassikcat** [openspeech-team/openspeech#148]
# ❓ Questions & Help

안녕하세요, openspeech의 코드를 보던 도중 궁금한 부분이 있어 질문드립니다.

Conv2d_extractor의 96번째 줄 이후 forward를 보면 input mel-spectrogram을 transpose(2, 3)한 뒤 

```python
batch_size, channels, dimension, seq_lengths = outputs.size() # line 103
```

처럼 batch, channel, dim, seq_len의 숫자를 구해주는 프로세스가 나옵니다.

그런데, torchaudio나 openspeech에서 사용하는 librosa의 spectrogram은 [channel, dim, seq_len] 형태로 나오기 때문에 [channel, seq_len, dim 형태가 될 것 같은데요, 실제로 [channels, dim, seq_len] 형태로 랜덤한 float텐서를 만들어 encoder 모듈을 돌려봤더니 텐서곱을 할 수 없다는 에러가 떴습니다.

그렇다면 혹시 line 103처럼 구해주신 이유가 openspeech에서 사용하는 데이터셋이 단순히 음성 데이터를 [channel, seq_len, dim] 형태로 반환하기 때문인지, 아니면 다른 이유가 있는 것인지 궁금합니다. 

감사합니다.

## Details
[openspeech](https://github.com/openspeech-team/openspeech/tree/main/openspeech)/[modules](https://github.com/openspeech-team/openspeech/tree/main/openspeech/modules)/conv2d_extractor.py
`line 96 - 107`
```python
    def forward(self, inputs: Tensor, input_lengths: Tensor) -> Tuple[Tensor, Tensor]:
        r"""
        inputs: torch.FloatTensor (batch, time, dimension)
        input_lengths: torch.IntTensor (batch)
        """
        outputs, output_lengths = self.conv(inputs.unsqueeze(1).transpose(2, 3), input_lengths)

        batch_size, channels, dimension, seq_lengths = outputs.size()
        outputs = outputs.permute(0, 3, 1, 2)
        outputs = outputs.view(batch_size, seq_lengths, channels * dimension)

        return outputs, output_lengths
```

오류

`[In]`: 
```python
import torch
from openspeech.encoders import ConformerEncoder 

tensor = torch.rand([1, 80, 273])
forward = ConformerEncoder(num_classes=12, encoder_dim=32)
result = forward(tensor, torch.tensor([273 for _ in range(tensor.shape[0])]))
```

`[out]`: `mat1 and mat2 shapes cannot be multiplied (19x2144 and 608x32)`

**Comment by upskyy** [comment]
안녕하세요, @Klassikcat 
openspeech에서 모든 inputs의 shape은 (batch, time, dimension) 형태를 가지고 있습니다. 그래서 `tensor = torch.rand([1, 273, 80])`로 넣어주셔야 합니다. 추가로 더 궁금하신 점은 말씀해주세요!
